### PR TITLE
Update pyne

### PR DESCRIPTION
**Pull request by bam241**
- starting to work toward multi-option build

- updating compiler requirement 
- adding host package requirement

### CODE REVIEW

**Comment by bam241** [comment]
also added build naming and number as variable

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
+{% set dagmc = dagmc or 'nodagmc' %}
+
+
+{% if moab != 'nomoab' %}
+{% set prefix = "moab_" + mpi %}
+{% else %}
+{% set prefix = "nomoab" %}
+{% endif %}
+
+{% if moab != 'nodagmc' %}
```
Comment (opotowsky) [inline recipe/meta.yaml]: do you mean 'dagmc' instead of 'moab' here?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,33 +34,37 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 0
+  number: {{ build }}
+  string: "{{ prefix }}}_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ build }}"
   # note that builds timeout on mac with many versions of python & numpy
-  skip: True  # [win or (osx and py34)]
+  skip: True  # [win or (osx and py34) or ((moab == 'nomoab) and (dagmc == 'dagmc'))]
```
Comment (opotowsky) [inline recipe/meta.yaml]: I think you're missing a close quote on 'nomoab'

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
 {% set variant = "openblas" %}
 
+# ensure moab is defined (needed for conda-smithy recipe-lint)
+{% set moab = moab or 'nomoab' %}
+# ensure dagmc is defined (needed for conda-smithy recipe-lint)
+{% set dagmc = dagmc or 'nodagmc' %}
+
+
+{% if moab != 'nomoab' %}
+{% set prefix = "moab_" + mpi %}
```
Comment (opotowsky) [inline recipe/meta.yaml]: is it ok to use double quotes? I can comb through for consistency in my follow-up  PR (so don't wory about it) -- just checking

**Review by opotowsky — COMMENTED** [review]
Just a few questions

File: recipe/meta.yaml (hunk 3 above)
Comment (bam241) [reply recipe/meta.yaml]: that's what they did in the Moab recipe :s don't know if they are important....
probably not :)

**Comment by bam241** [comment]
@opotowsky fixed ! :)

**Comment by kkiesling** [comment]
I saw this was merged, but wouldn't it be better to just get a clean PyNE lite version first without the pyMOAB and DAGMC dependencies at all? So we can make one PR that is only PyNE that works and then make a second PR that includes the DAGMC and MOAB dependencies. The second one depends on have a DAGMC conda forge, does it not? It would probably be better to make those two separate PRs right?

**Pull request merged**